mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -561,6 +588,177 @@ class WidgetAdapter(
         }
     }
 
+    class InputViewHolder internal constructor(initData: ViewHolderInitData) : LabeledItemBaseViewHolder(
+        initData,
+        R.layout.widgetlist_inputitem,
+        R.layout.widgetlist_inputitem_compact
+    ) {
+        private val inputTextLayout: TextInputLayout = itemView.findViewById(R.id.widgetinput)
+        private val inputText: TextInputEditText = itemView.findViewById(R.id.widgetinputvalue)
+        private var isBinding = false
+        private var hasChanged = false
+
+        private var updateJob: Job? = null
+        private var oldValue: String? = null
+
+        init {
+            inputText.doAfterTextChanged { if (!isBinding) hasChanged = true }
+            inputText.setOnFocusChangeListener { view, hasFocus ->
+                if (!hasFocus) {
+                    hideKeyboard(view)
+                    if (hasChanged) updateValue()
```

Comment:
Didn't we want to require explicit confirmation (via IME ACTION_DONE)? If so, I'd rather revert to the previous value here.

Reply:
I intended it also to just work on a focus change, but I will change.